fuzzypp/src/cliopts.hpp

```diff
@@ -0,0 +1,51 @@
+#ifndef FUZZYPP_CLIOPTS_HPP
+#define FUZZYPP_CLIOPTS_HPP
+
+#include <filesystem>
+#include <optional>
+#include <string>
+#include <vector>
+
+#include <cxxopts.hpp>
+
+namespace fuzzypp::cliopts {
+    class CliOptions {
+        public:
+            const std::vector<std::string> files;
+            const std::vector<std::string> include_files;
+            const std::vector<std::string> include_paths;
+            const std::vector<std::string> defines;
+            const std::vector<std::string> undefines;
+            const std::string output_directory;
```

Comment:
I assume this is required by the `cxxopts` library?

Reply:
That's right, I parse the options using `cxxopts` and put the results into this data structure.